File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        boolean flag = true;
        s = s.toLowerCase();
        for(int i=0, j=s.length()-1; i<=j;)
        {
            if(Character.isLetterOrDigit(s.charAt(i)) && Character.isLetterOrDigit(s.charAt(j))) {
                if(s.charAt(i) != s.charAt(j)) {
                    flag = false;
                    break;
                }
                i++;
                j--;
            }
            else if(!Character.isLetterOrDigit(s.charAt(i)))
                i++;
            else if(!Character.isLetterOrDigit(s.charAt(j)))
                j--;
        }
        return flag;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        flag = True
        i = 0
        j = len(s) - 1
        s = s.lower()

        while i<=j:
            if s[i].isalnum() and s[j].isalnum():
                if s[i] != s[j]:
                    flag = False
                    break
                i += 1
                j -= 1
            elif not s[i].isalnum():
                i += 1
            elif not s[j].isalnum():
                j -= 1
        
        return flag